src/journal/src/Journal.hs

```diff
@@ -87,19 +87,19 @@ stopJournal jour jc = do
 
 -- * Production
 
--- NOTE: pre-condition: `0 < BS.length bs <= maxByteSize`
 appendBS :: Journal -> ByteString -> IO ()
-appendBS jour bs = assert (0 < BS.length bs &&
-                           hEADER_SIZE + BS.length bs <= jMaxByteSize jour) $ do
+appendBS jour bs = do
+  assertM (0 < BS.length bs &&
+           hEADER_SIZE + BS.length bs + fOOTER_SIZE <= jMaxByteSize jour)
```

Comment:
```suggestion
  assertM (0 < BS.length bs)
  assertM (hEADER_SIZE + BS.length bs + fOOTER_SIZE <= jMaxByteSize jour)
```
I think it is better to have two separate asserts instead of `&&` since if it fails it will point to which one failed.